templates/CRM/Member/Form/PaymentPlanToggler.tpl

```diff
@@ -91,16 +91,27 @@
             }
           });
         }
+        assignReceiveDate();
       });
 
       $('#payment_plan_schedule, #payment_instrument_id, #start_date, #end_date').change(() => {
         if ($('#payment_plan_schedule_row').is(":hidden")) {
           return;
         }
         generateInstalmentSchedule(isPriceSetSelected());
+        assignReceiveDate();
       });
     }
 
+    /**
+     * Assigns payment plan received date from either start date or join date
```

Comment:
this assigns first contribution receive date and not the payment plan, can you also change the function name to assignFirstContributionReceiveDate